src/misc/programs.cpp

```diff
@@ -170,75 +189,52 @@ bool Program::SuppressWriteOut(const char *format)
 	return true;
 }
 
-// For "\n" to "\r\n" expansion (0xA to OxD 0xA) in WriteOut* functions
-static char last_written_character = '\n';
+// TODO: Refactor messages and related functions like WriteOut that use variadic
+// args to instead use format strings when dosbox-staging starts using C++20
```

Comment:
For C++20 : do you want to explore this transition?

We want to switch as soon as C++20 compilers are available on all supported platforms. So the task is partly assessing if it's possible (in the near term) and then making the changes to update to it.

Reply:
That's great!

With Ubuntu 18.04 rolling off in April, the challenge will be supporting Ubuntu 20.04 : @bmunger , do you know if GCC 13.x will be available?

As for debian, ideally we could keep supportinf Buster 10.x, as it doesn't go EOL till 2024.

I think our sticking point are going to the Raspberry Pi (and if software stacks  like RetroPie can support GCC 13 -- @nemo93 , would you be able to enquire?)

We build the macOS add Windows binaries using Clang, however both have no issues with running latest versions.

`libfmt` might give us more breathing room, too.